pkg/cloud/cloud_test.go

```diff
@@ -624,3 +624,46 @@ func checkDaemonSetSelectorLabels(t *testing.T, ds *appsv1.DaemonSet, platformTy
 	assert.Equal(t, string(platformType), ds.Spec.Template.Labels[common.CloudNodeManagerCloudProviderLabel])
 	assert.Equal(t, string(platformType), ds.Spec.Selector.MatchLabels[common.CloudNodeManagerCloudProviderLabel])
 }
+
+func TestReplicas(t *testing.T) {
+	platforms := getPlatforms()
+	for platformName, platform := range platforms {
+
+		t.Run(platformName, func(t *testing.T) {
+			resources, err := GetResources(platform.getOperatorConfig())
+			assert.NoError(t, err)
+
+			for _, resource := range resources {
+				switch obj := resource.(type) {
+				case *appsv1.Deployment:
+					assert.Equal(t, derefReplicas(obj.Spec.Replicas), 2)
+				default:
+					// Nothing to check for
+				}
+			}
+		})
+
+		t.Run(fmt.Sprintf("%s single node", platformName), func(t *testing.T) {
+			cfg := platform.getOperatorConfig()
+			cfg.IsSingleReplica = true
+			resources, err := GetResources(cfg)
+			assert.NoError(t, err)
+
+			for _, resource := range resources {
+				switch obj := resource.(type) {
+				case *appsv1.Deployment:
+					assert.Equal(t, derefReplicas(obj.Spec.Replicas), 1)
+				default:
+					// Nothing to check for
+				}
+			}
+		})
+	}
+}
+
+func derefReplicas(num *int32) int {
+	if num != nil {
+		return int(*num)
+	}
+	return 1
+}
```

Comment:
Typically in a deref function you would have a second argument which is the default, might make it more usable later, wdyt?
```suggestion
func derefIntOrDefault(num *int32, default int32) int32 {
	if num != nil {
		return *num
	}
	return default
}
```